lib/web_hook_processor.rb

```diff
@@ -0,0 +1,60 @@
+require 'json'
+require 'octokit'
+require 'sinatra'
+require_relative 'ping_checker'
+require_relative 'signature_verifier'
+require_relative 'merge_checker'
+
+use Rack::Logger
+
+module Sinatra
+  helpers do
+
+    def logger
+      request.logger
+    end
+
+    class WebHookProcessor
```

Comment:
This is the main class that I'm delegating responsibility to in order to handle the response. It's in the `Sinatra` module because some tutorial said this is how I can access the `logger` and `halt` function etc (that tutorial was a liar!). I could also be misunderstanding how to delegate responsibility like this, and if anyone has suggestions or recommended reading, I would appreciate it :bow: